project/src/main/java/com/google/sps/servlets/QueryServlet.java

```diff
@@ -48,17 +53,38 @@ void init(PlacesFetcher inputFetcher) {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    ImmutableList<Place> filteredPlaces;
     try {
-      //TODO(M1): add call to filterer
-      response.setContentType("application/json");
-      response.getWriter().write(new Gson().toJson(
-          Places.randomSort(fetcher.fetch())
-          .stream()
-          .limit(MAX_NUM_PLACES_TO_RECOMMEND)
-          .collect(Collectors.toList())
-      ));
+      UserPreferences userPrefs =
+          UserPreferences.builder()
+              .setMinRating(Float.parseFloat(request.getParameter("rating")))
+              .setMaxPriceLevel(Integer.parseInt(request.getParameter("price")))
+              .setOpenNow(Integer.parseInt(request.getParameter("open")) != 0)
+              .setLocation(getLatLngFromString(request.getParameter("location")))
+              .setCuisines(ImmutableList.copyOf(request.getParameter("cuisines").split(",")))
+              .build();
+      filteredPlaces = Places.filter(
+          fetcher.fetch(userPrefs) /* places */,
+          Integer.parseInt(request.getParameter("rating")) /* min rating */,
+          true /* filter if no website */,
+          true /* filter branches of same place */
+      );
     } catch (Exception e) {
-      //TODO(M1): handle errors - TBD
+      response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+          "Fetching from Google Places API encountered a problem");
+      return;
     }
+    response.setContentType("application/json");
+    response.getWriter().write(new Gson().toJson(
+        Places.randomSort(filteredPlaces)
+            .stream()
+            .limit(MAX_NUM_PLACES_TO_RECOMMEND)
+            .collect(Collectors.toList())
+    ));
+  }
+
+  private LatLng getLatLngFromString(String coordinates) {
```

Comment:
static?

Reply:
Great, thanks. 
The reason you think it should be static is that it doesn't use any class variables and that it has a small and a "utility like" job? (I'm asking to acquire intuition for deciding that)

Reply:
Correct! I think it doesn't have any logic, but it's a small utility class.